File: design-patterns/ts/visitor/hosting-cost/container-service.ts
// container-service.ts

import HostingCalculatorVisitor from "./hosting-calculator-visitor";
import Service from "./service";

class ContainerService implements Service {

    private readonly price: number = 5.60;
    private quantity: number;

    constructor(quantity: number) {
        this.quantity = quantity;
    }

    getPrice(): number {
        return this.price;
    }

    getQuantity(): number {
        return this.quantity;
    }

    accept(hostingCalculatorVisitor: HostingCalculatorVisitor): number {
        return hostingCalculatorVisitor.visit(this);
    }
}

export default ContainerService;

File: design-patterns/go/visitor/hosting_cost/container_service.go
// container_servie.go

package main

type ContainerService struct {
	price    float64
	quantity int
}

func NewContainerService(quantity int) (containerService *ContainerService) {
	containerService = &ContainerService{}
	containerService.price = 5.60
	containerService.quantity = quantity
	return
}

func (containerService *ContainerService) GetPrice() float64 {
	return containerService.price
}

func (containerService *ContainerService) GetQuantity() int {
	return containerService.quantity
}

func (containerService *ContainerService) Accept(hostingCalculatorVisitor HostingCalculatorVisitor) float64 {
	return hostingCalculatorVisitor.ContainerVisit(containerService)
}
